src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -591,9 +594,11 @@ public AddressBalance GetAddressBalance(string address)
 
             lock (this.lockObject)
             {
+                HdAddress hdAddress = null;
+
                 foreach (Wallet wallet in this.Wallets)
                 {
-                    HdAddress hdAddress = wallet.GetAllAddressesByCoinType(this.coinType).FirstOrDefault(a => a.Address == address);
+                    hdAddress = wallet.GetAllAddressesByCoinType(this.coinType).FirstOrDefault(a => a.Address == address);
```

Comment:
Maybe pass address into `GetAllAddressesByCoinType`, thereby creating an overload for the method?

Reply:
Not entirely sure that would make sense, given the name of the method and its presumed purpose.

Reply:
I think this overload would be confusing given the name of the function.

Reply:
`GetAllAddressesByCoinTypeAndAddress` ? LOL maybe that is too much. Anyway, all good then 👍